src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -406,7 +408,8 @@ public AzureVMAgentTemplate(
             boolean doNotUseMachineIfInitFails,
             boolean enableMSI,
             boolean enableUAMI,
-            String uamiID) {
+            String uamiID,
+            int maximumDeploymentSize) {
```

Comment:
there's no need to modify the constructor, just add a new setter annotated with `DataboundSetter`

Reply:
How would this impact the [AzureVMTemplateBuilder class](https://github.com/jenkinsci/azure-vm-agents-plugin/blob/master/src/main/java/com/microsoft/azure/vmagent/builders/AzureVMTemplateBuilder.java)? I wasn't sure what that class is used for.

Reply:
Historically there was a java API for people to use via groovy scripts to automate configuring this plugin this class was provided to make it easier to configure.

It is also used in the integration tests to configure some different scenarios.

I don't require updates to the `AzureVMTemplateBuilder` as using the configuration as code plugin is a much better way to manage this plugins configuration (but I won't block changes to it).

Jenkins databinding dictates that required fields are in the constructor and optional fields are done with setters, adding a field to the constructor will force all jcasc users to set it after upgrade

If you do want to keep the changes there you should create a temporary instance variable and call the new setter created for this field.

Reply:
Okay, it all makes sense now. I reset my changes to the `AzureVMTemplateBuilder` stuff. I also removed the parameter from the constructor and changed it to a DataBoundSetter.